code/part-two/client/source/Collection.jsx

```diff
@@ -42,13 +43,34 @@ export class Collection extends React.Component {
       });
   }
 
+  createCollection() {
+    submitPayloads(this.props.privateKey, { action: 'CREATE_COLLECTION' })
+      .then(() => {
+        this.fetchCollection(this.state.publicKey);
+      })
+      .catch(err => {
+        console.error(err);
+        alert('Something went horribly wrong!');
+      });
+  }
+
   render() {
     console.log('RENDERING: <Collection />');
     const { publicKey, collection } = this.state;
     if (!collection) {
       return (
         <div>
-          We can't find anything for a collection at <code>{publicKey}</code>!
+          We can't find a collection for public key <code>{publicKey}</code>!
+          <br /><br />
+          { // if user is trying to view their non-existing collection
+            this.props.publicKey === publicKey && (
+              <span>
+                If you are logged in as the owner of this collection and would
+                like to create a new collection, please click {' '}
+                <a href="#" onClick={this.createCollection}>here</a>!
```

Comment:
What is the thinking behind creating a collection under the View Collection page, instead of doing it as part of the signup process? Surely if I'm a user, the only reason I am generating a private key is to also create a collection.